dev/DynamicDependency/WinRTModuleManager.cpp

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "WinRTModuleManager.h"
+
+std::mutex MddCore::WinRTModuleManager::s_lock;
+std::vector<std::shared_ptr<MddCore::WinRTPackage>> MddCore::WinRTModuleManager::s_winrtPackages;
+
+bool MddCore::WinRTModuleManager::GetThreadingType(
+    HSTRING className,
+    ABI::Windows::Foundation::ThreadingType& threadingType)
+{
+    auto lock{ std::unique_lock<std::mutex>(s_lock) };
+
+    auto threadingModel{ MddCore::WinRTModuleManager::GetThreadingModel(className) };
+    if (threadingModel == MddCore::WinRT::ThreadingModel::Unknown)
+    {
+        return false;
+    }
+
+    THROW_IF_FAILED(ToThreadingType(threadingModel, threadingType));
+    return true;
+
+}
+
+MddCore::WinRT::ThreadingModel MddCore::WinRTModuleManager::GetThreadingModel(
+    HSTRING className)
+{
+    if (!s_winrtPackages.empty())
+    {
+        std::wstring activatableClassId{ WindowsGetStringRawBuffer(className, nullptr) };
+        for (auto& winrtPackage : s_winrtPackages)
```

Comment:
Do you think we will hit scale problems on these linear searches?

Reply:
No. We don't have a single list. For proper search order semantics we have 1 list per package, which itself is broken up into 1 list per `<Extension Category="windows.inProcessServer"...>`

In theory [worst case for a single `<Extension>` is 65535 ActivatableClassIds](https://docs.microsoft.com/uwp/schemas/appxpackage/uapmanifestschema/element-inprocessserver).

I thought about using a map or at least sorting the list and doing a binary search but that seemed a premature optimization and easily addressed later.  Will make a note for follow up. Thanks for reminding me.

Reply:
OK, thanks. :-)